src/main/java/eu/dissco/backend/repository/ElasticSearchRepository.java

```diff
@@ -186,19 +191,21 @@ public Pair<Long, List<DigitalSpecimen>> search(Map<String, List<String>> params
     return getDigitalSpecimenSearchResults(searchRequest);
   }
 
-  public Pair<Long, List<DigitalSpecimen>> getSpecimens(int pageNumber, int pageSize) throws IOException {
+  public Pair<Long, List<DigitalSpecimenWrapper>> getSpecimens(int pageNumber, int pageSize)
+      throws IOException {
     var offset = getOffset(pageNumber, pageSize);
     var pageSizePlusOne = pageSize + ONE_TO_CHECK_NEXT;
     var searchRequest = new SearchRequest.Builder()
-        .index(DIGITAL_SPECIMEN_INDEX)
+        .index(properties.getDigitalSpecimenIndex())
         .trackTotalHits(t -> t.enabled(Boolean.TRUE))
         .from(offset)
         .size(pageSizePlusOne)
         .build();
     return getDigitalSpecimenSearchResults(searchRequest);
   }
 
-  private Pair<Long, List<DigitalSpecimen>> getDigitalSpecimenSearchResults(SearchRequest searchRequest) throws IOException{
+  private Pair<Long, List<DigitalSpecimenWrapper>> getDigitalSpecimenSearchResults(
+      SearchRequest searchRequest) throws IOException {
     var searchResult = client.search(searchRequest, ObjectNode.class);
     var totalHits = searchResult.hits().total().value();
```

Comment:
Add null check?

Reply:
Ok, we have control over whether this is null or not. But this does make the method more generic
"Total hit count information, present only if track_total_hits wasn't false in the search request."